.github/workflows/build.yaml

```diff
@@ -0,0 +1,39 @@
+name: build
+
+on: [push, pull_request]
+jobs:
+  build:
+    name: build
+    strategy:
+      matrix:
+        go-version: [1.16, 1.17, 1.18]
+        os: [ubuntu-latest]
```

Comment:
Do we need os in the matrix, will we be adding another os?

Reply:
not needed atm, removing